libtransmission/bitfield.cc

```diff
@@ -157,37 +154,35 @@ size_t Bitfield::countBits() const
     return this->true_count_;
 }
 
-void Bitfield::setBitsInArray(uint8_t* array, size_t bit_count)
+void Bitfield::setBitsInArray(std::vector<uint8_t> &array, size_t bit_count)
 {
-    uint8_t const val = 0xFF;
-    size_t const n = getStorageSize(bit_count);
+    size_t const byte_count = getStorageSize(bit_count);
 
-    if (n > 0)
+    if (byte_count > 0)
     {
-        memset(array, val, n - 1);
+        std::fill(array.begin(), array.begin() + byte_count - 1,  0xFF);
 
-        array[n - 1] = val << (n * 8 - bit_count);
+        array[byte_count - 1] = 0xFF << (byte_count * 8 - bit_count);
     }
 }
 
-void* Bitfield::getRaw(size_t* byte_count) const
+std::vector<uint8_t> Bitfield::getRaw() const
 {
     TR_ASSERT(this->bit_count_ > 0);
 
     size_t const n = getStorageSize(this->bit_count_);
-    uint8_t* newBits = tr_new0(uint8_t, n);
+    std::vector<uint8_t> newBits(n);
 
-    if (this->alloc_count_ != 0)
+    if (!this->bits_.empty())
     {
-        TR_ASSERT(this->alloc_count_ <= n);
-        std::memcpy(newBits, this->bits_, this->alloc_count_);
+        TR_ASSERT(this->bits_.size() <= n);
+        std::copy(this->bits_.cbegin(), this->bits_.cend(), std::back_inserter(newBits));
```

Comment:
need to add `#include <iterator> // std::back_inserter`

Reply:
Seems to not build on Windows without this, i'll add anyway